### PR TITLE
Auto OCP-29299

### PR DESCRIPTION
**Pull request by huiran0826**
https://privatebin-it-iso.int.open.paas.redhat.com/?0f8ec5c58b061cd5#IBm9d1js6f+7ngir4VfHDj63bxoho/0m78Jk0wps5Ko=

@zhaozhanqi @weliang1 @rbbratta @anuragthehatter 

Original PR was closed due to some messy files, open this new PR.

### CODE REVIEW

File: features/networking/sdn.feature [hunk 1]
```diff
@@ -358,3 +358,36 @@ Feature: SDN related networking scenarios
     | nmcli \| grep veth \| wc -l |
   And evaluation of `@result[:response].split("\n")[0]` is stored in the :no_of_veths clipboard
   Then the expression should be true> cb.no_of_veths==cb.no_of_unmanaged_veths
+
+  # @author huirwang@redhat.com
+  # @case_id OCP-29299
+  @admin
```
Comment (anuragthehatter) [inline features/networking/sdn.feature:364]: LGTM except this seems to be a @destructive candidate due to CNO patching involved irrespective of whether the behavior passes or fail. Thanks @huiran0826

File: features/networking/sdn.feature [hunk 2]
```diff
@@ -358,3 +358,36 @@ Feature: SDN related networking scenarios
...
+    When I run the :annotate client command with:
+       | resource     | network.operator.openshift.io                       |
+       | resourcename | cluster                                             |
+       | keyval       | networkoperator.openshift.io/network-migration=true |
+    Then the step should fail
+    And the output should contain:
+      | networks.operator.openshift.io "cluster" is forbidden |
+    Given I switch to cluster admin pseudo user
+    And as admin I successfully merge patch resource "network.config.openshift.io/cluster" with:
+      | {"spec":{"networkType":"OVNKubernetes"}}  |
```
Comment (weliang1) [inline features/networking/sdn.feature:375]: @huiran0826 Before trying to migrating to OVN, should test case make sure the origin networktype running in the cluster is "OpenShiftSDN"?

File: features/networking/sdn.feature (hunk 1 above)
Comment (huiran0826) [reply features/networking/sdn.feature:364]: Thanks @anuragthehatter , updated it.

File: features/networking/sdn.feature (hunk 2 above)
Comment (huiran0826) [reply features/networking/sdn.feature:376]: @weliang1  Thanks,  in the test case we already has the network backbend as  OpenShiftSDN.

File: features/networking/sdn.feature [hunk 3]
```diff
@@ -358,3 +358,37 @@ Feature: SDN related networking scenarios
...
+    When I run the :get client command with:
+      | resource | pods                               |
+      | o        | jsonpath={.items[*].metadata.name} |
+    Then the step should succeed
+    And evaluation of `@result[:response]` is stored in the :network_operator_pod clipboard
+    When I run the :logs client command with:
+      | resource_name | <%= cb.network_operator_pod %> |
+      | since         | 30s                            |
+    And the output should contain:
+      | Not applying unsafe change: invalid configuration|
```
Comment (pruan-rht) [inline features/networking/sdn.feature:394]: @huiran0826 minor nitpick... need space before then ending `|`.

Comment (huiran0826) [reply features/networking/sdn.feature]: Thanks @pruan-rht, fixed it.

**Pull request merged**